sycl/include/sycl/feature_test.hpp.in

```diff
@@ -30,7 +30,7 @@ __SYCL_INLINE_VER_NAMESPACE(_V1) {
 
 // TODO: Move these feature-test macros to compiler driver.
 #define SYCL_EXT_INTEL_DEVICE_INFO 6
-#define SYCL_EXT_INTEL_DEVICE_ARCHITECTURE 1
+#define SYCL_EXT_ONEAPI_DEVICE_ARCHITECTURE
```

Comment:
Thanks for fixing the name of the macro!

Reply:
Shouldn't we set value to `1` as we do for all other extensions?

Reply:
It's decided to remove the API versioning entirely as API versioning is required for "supported" extensions, but not for "experimental".

Reply:
What @mmoadeli says is true, but we should probably `#define SYCL_EXT_ONEAPI_DEVICE_ARCHITECTURE 1` to be consistent with the other extensions that are experimental.

Reply:
Agree. This inconsistence will make the usage of these macro complex/messy.

Reply:
@gmlueck @bader I am more in favour of not having versioning for experimental extensions. But, it's  fine to me to put versioning back if you believe it's better overall.

Reply:
I think we should keep the wording you added to the extension specification saying it is not versioned.  However, we should still define the macro to `1`.  This is consistent with the wording that says:

```
[%header,cols="1,5"]
|===
|Value
|Description

|1
|The APIs of this experimental extension are not versioned, so the
 feature-test macro always has this value.
|===
```

Note that the specification says that the macro has the value `1`.

Reply:
> I think we should keep the wording you added to the extension specification saying it is not versioned. However, we should still define the macro to `1`. This is consistent with the wording that says:
> 
> ```
> [%header,cols="1,5"]
> |===
> |Value
> |Description
> 
> |1
> |The APIs of this experimental extension are not versioned, so the
>  feature-test macro always has this value.
> |===
> ```
> 
> Note that the specification says that the macro has the value `1`.

@gmlueck thank you. Done.